pkg/descheduler/strategies/duplicates.go

```diff
@@ -210,10 +210,7 @@ func RemoveDuplicatePods(
 				// It's assumed all duplicated pods are in the same priority class
 				// TODO(jchaloup): check if the pod has a different node to lend to
 				for _, pod := range pods[upperAvg-1:] {
-					if _, err := podEvictor.EvictPod(ctx, pod, nodeMap[nodeName], "RemoveDuplicatePods"); err != nil {
-						klog.ErrorS(err, "Error evicting pod", "pod", klog.KObj(pod))
-						break
-					}
+					podEvictor.EvictPod(ctx, pod)
```

Comment:
It appears that functionality is being changed here? In other words, before it would break on a single "failed" eviction but now it loops through all pods.

Should it be preserved?
```suggestion
					if !podEvictor.EvictPod(ctx, pod) {
						break
					}

Reply:
In the past there was only a single error issued (exceeding limit of number of pods allowed to be evicted per node). So it made sense to break and move to another node. However, few months back we added a [limit](https://github.com/kubernetes-sigs/descheduler/pull/658) for the number of pods evicted per namespace. So when the namespace limit gets exceeded we can still continue evicting.

Though, I plan to introduce a check for the node limit exceeded. I will incorporate it in this PR once https://github.com/kubernetes-sigs/descheduler/pull/847 is merged.

Reply:
Just introduced `NodeLimitExceeded` method for performing the check.